app/portainer/services/notifications.js

```diff
@@ -14,28 +24,52 @@ angular.module('portainer.app').factory('Notifications', [
       toastr.warning($sanitize(text), $sanitize(title), { timeOut: 6000 });
     };
 
-    service.error = function (title, e, fallbackText) {
+    service.error = function (title, e, fallbackText, APIType = service.ApiTypes.KUBERNETES) {
       var msg = fallbackText;
-      if (e.err && e.err.data && e.err.data.message) {
-        msg = e.err.data.message;
-      } else if (e.err && e.err.data && e.err.data.details) {
-        msg = e.err.data.details;
-      } else if (e.data && e.data.details) {
-        msg = e.data.details;
-      } else if (e.data && e.data.message) {
-        msg = e.data.message;
-      } else if (e.data && e.data.content) {
-        msg = e.data.content;
-      } else if (e.message) {
-        msg = e.message;
-      } else if (e.err && e.err.data && e.err.data.length > 0 && e.err.data[0].message) {
-        msg = e.err.data[0].message;
-      } else if (e.err && e.err.data && e.err.data.err) {
-        msg = e.err.data.err;
-      } else if (e.data && e.data.err) {
-        msg = e.data.err;
-      } else if (e.msg) {
-        msg = e.msg;
+      if (APIType === service.ApiTypes.DOCKER) {
```

Comment:
There is a lot of repetition in conditions on both sides of the APITypes, we could either extract them to a function to return the `msg` or/and separate the specific 2 cases